lib/dependabot/file_parsers/go/go_mod_parser.rb

```diff
@@ -0,0 +1,93 @@
+# frozen_string_literal: true
+
+require "dependabot/dependency"
+require "dependabot/file_parsers/base/dependency_set"
+require "dependabot/file_parsers/python/pip"
+require "dependabot/errors"
+
+module Dependabot
+  module FileParsers
+    module Go
+      class GoModParser
+        def initialize(dependency_files:)
+          @dependency_files = dependency_files
+        end
+
+        def dependency_set
+          dependency_set = Dependabot::FileParsers::Base::DependencySet.new
+          dependency_set += go_mod_dependencies
+          dependency_set
+        end
+
+        private
+
+        attr_reader :dependency_files
+
+        def go_mod_dependencies
+          dependencies = Dependabot::FileParsers::Base::DependencySet.new
+
+          i = 0
+          chunks = module_info(go_mod).lines.group_by do |line|
+            if line == "{\n"
+              i += 1
+            else
+              i
+            end
+          end
+          deps = chunks.values.map { |chunk| JSON.parse(chunk.join) }
+
+          deps.each do |dep|
+            # The project itself appears in this list as "Main"
+            next if dep["Main"]
+
+            reqs = [{
+              requirement: dep["Indirect"] ? nil : dep["Version"],
```

Comment:
The requirements array is set to `[]` if this is an indirect requirement, so this check isn't required.